src/components/Profile/Profile.js

```diff
@@ -3,52 +3,60 @@ import './Profile.css';
 import { connect } from 'react-redux';
 
 class Profile extends Component {
     constructor(props) {
         super(props);
-        this.memberID = props.memberID ;
-        this.mock_reservations = props.mock_reservations;
+        /** @type { String } */
+        this.memberID = props.memberID;
+        /** @type { number } */
+        this.user_index_returned = ReturnedComponents.records.findIndex( 
+            reservation => reservation.memberID === this.memberID 
+        );
+        /** @type {!Array<{componentID: String, quantity: number, dateReturned: String}>, ...}>}*/
+        this.returned_components = (
+            this.user_index_returned === -1? 
+            null : 
+            ReturnedComponents.records[this.user_index_returned].returnedComponents
+        );
+        
+        this.handleChangeReturned = this.handleChangeReturned.bind(this);
         this.loadReservations = this.loadReservations.bind(this);
-    }
+        this.loadReturned = this.loadReturned.bind(this);
+        this.mock_reservations = props.mock_reservations;
 
-    handleModalState() {
-        this.modalEnabled = !this.modalEnabled;
+        this.state = {
+            /** @type { number } */
+            returned_user_index: this.user_index_returned,
+            /** @type {!Array<{componentID: String, quantity: number, dateReturned: String}>, ...}>}*/
+            returnedComponents: this.returned_components
+        }
     }
-
-    loadReservations() {
-        const componentsList = [];
-        let countComponents = 0;
-        
-        for (let reservations = 0; reservations < this.mock_reservations.length; reservations++) {
-            const reservationArray = this.mock_reservations[reservations].reservation;
-            for (let i = 0; i < reservationArray.length; i++) {
-                const style = (countComponents % 2 === 0) ? "oddRow" : "evenRow";
-                componentsList.push(
-                    <div>
-                        <Row className={ style }>
-                            <Col className="reservations">{this.mock_reservations[reservations].date}</Col>
-                            <Col className="reservations">{reservationArray[i].componentID}</Col>
-                            <Col className="reservations">{reservationArray[i].quantity}</Col>
-                        </Row>
-                    </div>
-                );
-                countComponents++;
-            }
-
+    
+    /*
+    This function is used to track the change when user clicks in modal button
+    */
+    handleChangeReturned() {
+        if(this.state.returned_user_index === -1) {
+            this.user_index_returned = ReturnedComponents.records.findIndex(
+                 reservation => reservation.memberID === this.memberID 
+            );
         }
-        return componentsList;
+        this.returned_components = ReturnedComponents.records[this.user_index_returned].returnedComponents;
```

Comment:
This code is repeated right? Consider declaring it as a function as mentioned in a comment above.